test/client.js

```diff
@@ -1063,3 +1063,38 @@ test('emit disconnect after destory', t => {
     })
   })
 })
+
+test('parser dinamic allocation', { only: true }, t => {
+  t.plan(3)
```

Comment:
Should we go deeper and actually check the parser `bufferSize` property?

Reply:
No I don't think that is necessary. But, wouldn't hurt either.

Reply:
I'll add it in https://github.com/nodejs/undici/pull/704